### PR TITLE
Improvements to config subcommand

### LINKED ISSUE #286: Configure subcommand does not support "--custom-targets-json"
**Describe the bug**
It's currently not possible to use the "configure" sub command with "out-of-source" custom targets because it doesn't support the "--custom-targets-json" option.

**To Reproduce**
1. Run "mbed-tools configure --custom-targets-json /some/path -m K64F -t GCC_ARM"
2. See error

**Expected behavior**
It should generate an mbed_config.cmake file, just like "mbed-tools compile" does.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: Linux
 - Version: Ubuntu 20.04

### LINKED ISSUE #287: Allow setting custom output directory in "configure" sub command
**Is your feature request related to a problem? Please describe.**
I use a custom build tool (that calls cmake under the hood) to build packages in my project. The fact mbed-tools use such an unusual build directory makes things awkward to integrate. 

**Describe the solution you'd like**
It would be great if mbed-tools could allow passing an "-o/--output-dir" option to configure that would allow us to customize where mbed_config.cmake will be written to.

**Describe alternatives you've considered**
Heuristically trying to find out where mbed_config.cmake was generated and moving to an appropriate location. This solution is suboptimal, at best.

### PR DESCRIPTION
**Pull request by g-arjones**
### Description

Fixes #287
Fixes #286 

### Test Coverage

- [x]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

**Comment by rwalton-arm** [comment]
Thanks for the PR. Could you squash the commit "Fix flake8 linter offenses" so the history is clean? We merge the patchset as is, and we'd prefer not to have "fixup" commits in the history. You'll also need to rebase as the branch is out of date with the base branch.

**Comment by g-arjones** [comment]
No problem. Updated...

File: news/20210525103622.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Properly handle --custom-targets-json in configure sub command
```
Comment (Patater) [inline news/20210525103622.bugfix:1]: Could you please rename this file to `286.bugfix`?

File: news/20210525104111.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Allow choosing output directory in configure sub command
```
Comment (Patater) [inline news/20210525104111.bugfix:1]: Please rename this to `287.bugfix`. Thanks!

File: tests/cli/test_configure.py [hunk 3]
```diff
@@ -23,3 +25,13 @@ def test_generate_config_called_with_mbed_os_path(self, program, generate_config
...
+
+    @mock.patch("mbed_tools.cli.configure.generate_config")
+    @mock.patch("mbed_tools.cli.configure.MbedProgram")
+    def test_custom_targets_location_used_when_passed(self, program, generate_config):
+        program = program.from_existing()
+        custom_targets_json_path = pathlib.Path("custom", "custom_targets.json")
+        CliRunner().invoke(configure, ["-t", "gcc_arm", "-m", "k64f", "--custom-targets-json", custom_targets_json_path])
+
+        generate_config.assert_called_once_with("K64F", "GCC_ARM", program)
+        self.assertEqual(program.files.custom_targets_json, custom_targets_json_path)
```
Comment (Patater) [inline tests/cli/test_configure.py]: GitHub is showing removal of newline at the end of the file. Could you please add it back in commit "config: Handle --custom-targets-json"?

**Review by Patater — CHANGES_REQUESTED** [review]
Please update the commit messages with an explanation of what motivates the change and any information about why your chosen method to solve the problem is preferable.

**Comment by g-arjones** [comment]
Done.

**Comment by g-arjones** [comment]
Did I miss anything?

**Review by rwalton-arm — COMMENTED** [review]
Looks good to me. Just needs to be rebased and then we can merge.

**Comment by g-arjones** [comment]
Rebased.

**Comment by rwalton-arm** [comment]
We had an automated release this morning and the branch is out of date again. Sorry! If you could rebase one more time we will get this in today.

**Comment by g-arjones** [comment]
Rebased, again.

**Review by rwalton-arm — APPROVED** [review]

**Pull request merged**